xla/service/gpu/parallel_loop_emitter.cc

```diff
@@ -183,9 +183,7 @@ ParallelLoopEmitter::EmitIndexAndSetExitBasicBlock(absl::string_view loop_name,
   // for that dimensions.  This helps LLVM generate vectorized codes
   // in that cases.
   llvm::Value* row_index = nullptr;
-  if (!launch_config_.row_vectorized) {
```

Comment:
As far as I understand, these changes are just a NFC refactoring, and are not related to the real changes being done in this PR? If yes, maybe split them into their own PR?
I would say the new code is not necessarily more easy to read. I agree with the changes to swap if/else blocks and get rid of the negation, but I would prefer if the loop still starts from 1 and we handle the i = 0 case outside the loop.

Reply:
Yes, this is a NFC refactoring, I can put it in another PR.

Reply:
Done. This PR only contains SPIR target/intrinsic now.